pkg/server/endpoints/node/handler.go

```diff
@@ -66,6 +66,8 @@ func NewHandler(config HandlerConfig) *Handler {
 func (h *Handler) Attest(stream node.Node_AttestServer) (err error) {
 	counter := telemetry_server.StartNodeAPIAttestCall(h.c.Metrics)
 	defer counter.Done(&err)
+	errCode := codes.OK
+	defer telemetry_common.AddErrorClass(counter, &errCode)
```

Comment:
as long as `err` is a named return value (which it is), the following is a bit simpler and less error prone than having to manually set `errCode` at each return:

```
defer func() {
    if code := status.Code(err); code != codes.OK {
        counter.AddLabel(telemetry.Error, code.String())
    }
}()
```

Reply:
You can wrap that up into a function that takes an *err if you want (similar to `Done`)

Reply:
would it be beneficial also to have all error returns actually be Status Code Errors? I noticed many are just general go errors.

Reply:
It isn't technically necessary since `codes.Internal` is used (and returned by `status.Code`) when the error is not a `status.Status`. If there are spots that would be expressed more clearly using a different status code, that seems fine to me.

Reply:
It seems to return `Unknown` as default, which was leading to interesting test errors on initial pass:
```
func Code(err error) codes.Code {
	// Don't use FromError to avoid allocation of OK status.
	if err == nil {
		return codes.OK
	}
	if se, ok := err.(interface {
		GRPCStatus() *Status
	}); ok {
		return se.GRPCStatus().Code()
	}
	return codes.Unknown
}
```

Reply:
Oh yep, you are right.